openapi.yaml

```diff
@@ -4075,14 +4085,16 @@ components:
         The billing plan to process and charge the job or service with.
 
         Billing plans MUST be accepted in a *case insensitive* manner.
-        If `null` is specified by the client explicitly,
-        the server MUST store the default plan and persist it regardless of any
-        changes to the default billing plan in the future. If given, the default
-        plan specific to the user has higher priority than the default plan of
-        the back-end.
-
-        The list of plans and the default plan for the back-end can be retrieved
-        from `GET /`. The default plan for the user can be retrieved from `GET /me`.
+        Back-ends MUST resolve the billing plan in the following order:
+        1. Use the `plan` that has been provided in the request, if a non-null value is present.
+
+        If `null` has been provided for the `plan`:
+        2. Use the `default_plan` exposed through `GET /me`, if available.
+        3. Otherwise, use the `default_plan` exposed through `GET /`.
+
+        The resolved plan MUST be persisted permanently, regardless of any 
```

Comment:
> MUST be persisted permanently

That sounds quite strong and vague at the same time. I guess this is just about explicitly associating a plan to a processing request (e.g. exposed in metadata)? Or is it also about legal traceability?

Reply:
Tried to clarify:

        The resolved plan MUST be persisted permanently so that the it is explicitly
        associated to a processing request and doesn't change when changes are made
        to the exposed billing plans in `GET /` in the future.

Reply:
Another thing is that "persist"  here is a bit tech-speak with a bit different meaning than the general verb "persist". I think the statement works just fine as 

> The resolved plan MUST be associated explicitly to ...

Reply:
We are in a developer resource here so persist should be fine especially as it's explained directly afterwards.